app/src/main/java/com/example/musicapp/MainActivity.kt

```diff
@@ -0,0 +1,209 @@
+package com.example.musicapp
+
+import android.Manifest
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import android.content.pm.PackageManager
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.os.IBinder
+import android.provider.MediaStore
+import android.view.View
+import android.widget.*
+import androidx.recyclerview.widget.LinearLayoutManager
+import kotlinx.android.synthetic.main.activity_main.*
+import java.util.Timer
+import java.util.TimerTask
+import kotlin.collections.ArrayList
+import kotlin.concurrent.thread
+
+data class Song(val id: Int, val title: String, val duration: Int)
+
+class MainActivity : AppCompatActivity(), MusicInterface.View, MusicInterface,
+    View.OnClickListener {
+    private val listSong: MutableList<Song> = ArrayList()
+
+    private lateinit var adapter: Adapter
+    private lateinit var mService: MusicService
+    private lateinit var musicPresenter: MusicPresenter
```

Comment:
K khuyến khích dùng late init var, a đọc thêm 1 số tài liệu về nó nhé
```suggestion
    private var adapter: Adapter? = null
    private var mService: MusicService? = null
    private var musicPresenter: MusicPresenter? = null
```

Reply:
3 hàm này sử dụng biến kế thừa not nullable nên có lẽ sử dụng lateinit thì vẫn hay hơn

Reply:
nghiên cứu dùng `by lazy`